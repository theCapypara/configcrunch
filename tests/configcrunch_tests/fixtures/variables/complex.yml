base:
  level_dict:
    a:
      name: "{{ parent().more.source }}"
      more:
        a: "{{ level_helper() }}"
        brackets_key: brackets_content
    b:
      name: "{{ parent().simple_helper() }}"
      more:
        a: "{{ parent().more.source|upper }}" # Jinja2 filter
        b_param_top_level: "{{ level_helper_taking_param('sun') }}"
  more:
    source: source
    a: "{{ level_dict.a.parent().more.source }}"
    brackets: "{{ level_dict.a.more['brackets_key'] }}"
    b: "{{ level_dict.a.parent().simple_helper() }}"
    c: "{{ level_dict.a.level_helper() }}"
    d: "{{ level_dict.a.parent().level_dict.b.level_helper() }}"
    e: "{{ level_dict.a.level_helper_taking_param('world') }}"
    e_in_dict:
      please_work: "does this {{ level_dict.a.level_helper_taking_param('world')}} work?"
      please_work_double_quotes: "does this {{ level_dict.a.level_helper_taking_param(\"househousehouse\")}} still work?"
    e_in_list: ["{{ level_dict.a.level_helper_taking_param('world') }}"]
