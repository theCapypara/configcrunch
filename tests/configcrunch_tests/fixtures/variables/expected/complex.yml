base:
  level_dict:
    a:
      $name: a
      name: source
      more:
        a: level
        brackets_key: brackets_content
    b:
      $name: b
      name: simple
      more:
        a: SOURCE
        b_param_top_level: "level_param: sun"
  more:
    source: source
    a: source
    brackets: brackets_content
    b: simple
    c: level
    d: level
    e: "level_param: world"
    e_in_dict:
      please_work: "does this level_param: world work?"
      please_work_double_quotes: "does this level_param: househousehouse still work?"
    e_in_list: ["level_param: world"]
